lib/bokken_web/controllers/file_controller.ex

```diff
@@ -55,12 +55,25 @@ defmodule BokkenWeb.FileController do
 
   def create(conn, %{"file" => file_params}) do
     user_id = conn.assigns.current_user.id
+    files = Documents.list_files(%{"user_id" => user_id})
+
+    total_size = Enum.reduce(files, 0, fn file, acc -> acc + File.file_size(file) end)
 
     with {:ok, %File{} = file} <- Documents.create_file(Map.put(file_params, "user_id", user_id)) do
-      conn
-      |> put_status(:created)
-      |> put_resp_header("location", Routes.file_path(conn, :show, file))
-      |> render("show.json", file: file)
+      file_size = File.file_size(file)
+
+      if total_size + file_size > 6_000_000 do
+        Documents.delete_file(file)
```

Comment:
This check should be inside `create_file` function. Imagine a case where the `delete_file` returns an error - we would stay with a created file.
I would suggest you to put some kinda of check on the changeset used by the `create_file` function. Since you also pass `user_id` to the function, it would be very easy to add a check on the changeset pipeline.

Ideally the _file size per user_ would be a column on the user's table, but we can take that to another PR.